### PR TITLE
Replace some `dynamic` to `Object?` type

### PR DESCRIPTION
**Pull request by xu-baolin**
Replace some `dynamic` to `Object?` type

### CODE REVIEW

**Comment by xu-baolin** [comment]
@Hixie hi, can this test-exempt?

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 1]
```diff
@@ -5621,10 +5621,10 @@ class _HistoryProperty extends RestorableProperty<Map<String?, List<Object>>?> {
 
   @override
   Map<String?, List<Object>>? fromPrimitives(Object? data) {
-    final Map<dynamic, dynamic> casted = data! as Map<dynamic, dynamic>;
-    return casted.map<String?, List<Object>>((dynamic key, dynamic value) => MapEntry<String?, List<Object>>(
+    final Map<Object?, Object?> casted = data! as Map<Object?, Object?>;
+    return casted.map<String?, List<Object>>((Object? key, Object? value) => MapEntry<String?, List<Object>>(
       key as String?,
-      List<Object>.from(value as List<dynamic>, growable: true),
+      List<Object>.from(value as List<Object?>, growable: true),
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/navigator.dart:5627]: This is causing the analyzer failure. I believe it's expecting a `List<Object>`, but you're giving it a `List<Object?>`

**Review by shihaohong — COMMENTED** [review]
The PR looks good, minus one analyzer failure.

I'm wondering if it really isn't possible to test for this change, since it'd be quite easy to just start using `dynamic` everywhere instead of `Object?` in the codebase in the future

File: packages/flutter/lib/src/widgets/navigator.dart (hunk 1 above)
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/navigator.dart:5627]:
```suggestion
      List<Object>.from(value as List<Object>, growable: true),
```

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 2]
```diff
@@ -5621,10 +5621,10 @@ class _HistoryProperty extends RestorableProperty<Map<String?, List<Object>>?> {
 
   @override
   Map<String?, List<Object>>? fromPrimitives(Object? data) {
-    final Map<dynamic, dynamic> casted = data! as Map<dynamic, dynamic>;
-    return casted.map<String?, List<Object>>((dynamic key, dynamic value) => MapEntry<String?, List<Object>>(
+    final Map<Object?, Object?> casted = data! as Map<Object?, Object?>;
+    return casted.map<String?, List<Object>>((Object? key, Object? value) => MapEntry<String?, List<Object>>(
       key as String?,
-      List<Object>.from(value as List<dynamic>, growable: true),
+      List<Object>.from(value as List<Object>, growable: true),
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/navigator.dart:5627]:
```suggestion
      List<Object>.from(value! as List<Object>, growable: true),
```

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 3]
```diff
@@ -5621,10 +5621,10 @@ class _HistoryProperty extends RestorableProperty<Map<String?, List<Object>>?> {
 
   @override
   Map<String?, List<Object>>? fromPrimitives(Object? data) {
-    final Map<dynamic, dynamic> casted = data! as Map<dynamic, dynamic>;
-    return casted.map<String?, List<Object>>((dynamic key, dynamic value) => MapEntry<String?, List<Object>>(
+    final Map<Object?, Object?> casted = data! as Map<Object?, Object?>;
+    return casted.map<String?, List<Object>>((Object? key, Object? value) => MapEntry<String?, List<Object>>(
       key as String?,
-      List<Object>.from(value as List<dynamic>, growable: true),
+      List<Object>.from(value! as List<Object>, growable: true),
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/navigator.dart:5627]:
```suggestion
      List<Object>.from(value! as List<Object?>, growable: true),
```

**Comment by Hixie** [comment]
test-exempt: code refactor without semantic change

**Review by shihaohong — APPROVED** [review]

**Pull request merged**